src/main/scala/com/github/takezoe/solr/scala/SolrClient.scala

```diff
@@ -37,9 +38,42 @@ class SolrClient(url: String)
       }
     }
   }
-  
+
+  /**
+    * Execute given operation in the transaction to a collection
+    *
+    * I'm not a huge fan of this function as you can still get an NPE, if your operations are not properly constructed.
+    * That said, support for the collection mode should be supported for all functionality.
+    *
+    * The transaction is committed if operation was successful.
+    * But the transaction is rolled back if an error occurred.
+    */
+  def withTransactionOnCollection[T](operations: => T, collection: String): T = {
```

Comment:
I prefer `withTransactionOnCollection[T](collection: String)(operations: => T): T = {...}` than yours because it allows us to write as below:
```scala
withTransactionOnCollection("collection"){
  ...
}
```
Another idea is `withTransactionOnCollection[T](collection: String = "")(operations: => T): T = {...}`. In this case, client cide will be as below. This is a breaking change, but I think it brings consistency and it's an acceptable change.
```scala
withTransaction(){
   ...
}

withTransaction("collection"){
  ...
}
```
Wha do you think?

Reply:
I'm fine with the first one, happy to accommodate the second, but for my PR, i was going to go with no breaking changes, as soon as i realized that could easily be done.  I've updated the code to use your first preference.  I'd be happy to work more on it, for now i think this is a good change and allows a publish to the original issue.  Thoughts?

Reply:
Thanks. Your change looks good to me! No breaking change is a good reason.